src/core/util/props.js

```diff
@@ -21,7 +21,7 @@ export function validateProp (
   const absent = !hasOwn(propsData, key)
   let value = propsData[key]
   // handle boolean props
-  if (getType(prop.type) === 'Boolean') {
+  if (getTypeList(prop.type).indexOf('Boolean') >= 0) {
```

Comment:
Since this function is pretty perf-sensitive, I'd suggest a `isBooleanType(prop.type)` function that:

- If `prop.type` is Array, iterate with `for` and return if any equals `'Boolean'`
- otherwise just check `=== 'Boolean'`.